src/main.rs

```diff
@@ -30,49 +30,52 @@ mod overhead_light;
 mod rgb_led;
 mod serial;
 
+static mut USB_ENDPOINT_MEMORY: [u32; 1024] = [0; 1024];
+
 #[entry]
 fn main() -> ! {
-    let mut cp =
-        cortex_m::peripheral::Peripherals::take().expect("failed to get cortex_m peripherals");
-    let dp = pac::Peripherals::take().expect("failed to get stm32 peripherals");
+    let cp = cortex_m::peripheral::Peripherals::take().expect("failed to get cortex_m peripherals");
+    let dp = stm32::Peripherals::take().expect("failed to get stm32 peripherals");
 
-    // Take ownership over the raw flash and rcc devices and convert them into the corresponding
+    // Take ownership over the raw devices and convert them into the corresponding
     // HAL structs.
     // RCC = Reset and Clock Control
-    let mut flash = dp.FLASH.constrain();
-    let mut rcc = dp.RCC.constrain();
+    let rcc = dp.RCC.constrain();
 
     // The various system clocks need to be configured to particular values
     // to work with USB - we'll set them up here.
     let clocks = rcc
         .cfgr
-        .use_hse(8.mhz()) // Use the High Speed External 8MHz crystal
+        .use_hse(25.mhz()) // Use the High Speed External 25MHz crystal
         .sysclk(48.mhz()) // The main system clock will be 48MHz
-        .pclk1(24.mhz())  // Use 24MHz for the APB1 (Advanced Peripheral Bus 1)
-        .freeze(&mut flash.acr);
-
-    assert!(clocks.usbclk_valid());
-
-    // Needed in order for MonoTimer to work properly
-    cp.DCB.enable_trace();
+        .require_pll48clk()
+        .freeze();
 
-    // Prepare the alternate function I/O registers
-    let mut afio = dp.AFIO.constrain(&mut rcc.apb2);
+    // assert!(clocks.usbclk_valid());
```

Comment:
uncomment?

Reply:
Unfortunately this call doesn't exist in the `stm32f4xx-hal` crate. I'll look into porting it.